util/parsenum.h

```diff
@@ -107,26 +108,38 @@ _Pragma("clang diagnostic pop")
  * For an unsigned integer variable ${x}, this can also be invoked as
  * PARSENUM_BASE(x, s, base), in which case the minimum and maximum values are
  * set to the limits of the unsigned integer type.
+ *
+ * For a floating-point variable ${x}, the ${base} must be 0.
  */
-#define PARSENUM_BASE3(x, s, b)						\
+#define PARSENUM_BASE3(x, s, base, _define_str)				\
```

Comment:
If we want the eventual `PARSENUM_EX` to handle both `PARSENUM` and `PARSENUM_EX`, and if we want the `ASSERT_FAIL()` line to give an accurate error message, then this macro needs to know whether it's been called as `PARSENUM` or `PARSENUM_EX`.

Reply:
PS: I used the prepended `_` to indicate that this is not a variable which the user supplies.